kernel/src/process/posix_thread/mod.rs

```diff
@@ -227,7 +174,12 @@ impl PosixThread {
     /// This method does not perform permission checks on user signals.
     /// Therefore, unless the caller can ensure that there are no permission issues,
     /// this method should be used to enqueue kernel signals or fault signals.
-    pub fn enqueue_signal(&self, signal: Box<dyn Signal>) {
+    pub fn enqueue_signal(&self, signal: Box<dyn Signal>, thread: &Thread) {
+        debug_assert!(core::ptr::eq(self, thread.as_posix_thread().unwrap()));
+        if thread.is_exited() {
```

Comment:
Sorry, I didn't realize that we needed to accept a new argument to check the thread status. If we need to accept a new argument, it needs a `debug_assert!` for correctness. Maybe the previous version is okay....